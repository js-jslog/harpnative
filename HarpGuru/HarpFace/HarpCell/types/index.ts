import type { TextStyle, ViewStyle } from 'react-native'

export type Coord = number

export type YXCoord = [Coord, Coord]

export type HarpCellProps = {
  readonly yxCoord: YXCoord
}

export type HarpCellStyles = {
  readonly cell: ViewStyle
  readonly note: TextStyle
  readonly modifier: TextStyle
}
