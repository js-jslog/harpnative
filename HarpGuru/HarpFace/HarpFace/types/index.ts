import { ViewStyle } from 'react-native'

export type HarpFaceStyles = {
  readonly facewrapper: ViewStyle;
  readonly face: ViewStyle;
}
