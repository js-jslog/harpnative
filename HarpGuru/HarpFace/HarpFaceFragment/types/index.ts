import type { ViewStyle } from 'react-native'

import type { HarpRowProps } from '../../HarpRow'

export type HarpFaceFragmentProps = Omit<HarpRowProps, 'yCoord'>

export type FragmentFacts = {
  readonly columnCount: number
  readonly rowCount: number
}

export type HarpFaceFragmentStyles = {
  readonly fragment: ViewStyle
}
