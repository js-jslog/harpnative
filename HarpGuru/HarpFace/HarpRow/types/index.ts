import type { ViewStyle } from 'react-native'

import type { HarpCellProps } from '../../HarpCell'

export type Coord = number

export type HarpRowProps = Pick<HarpCellProps, 'activeHarpStrata' | 'setActiveHarpStrata' | 'activeDisplayMode' | 'activeDisplayMode'> & {
  readonly yCoord: Coord
  readonly xRange: ReadonlyArray<number>
}

export type HarpRowStyles = {
  readonly row: ViewStyle
}
