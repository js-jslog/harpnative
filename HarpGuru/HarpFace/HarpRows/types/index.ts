import type { ReactElement } from 'react'

export type HarpRows = {
  readonly top: ReactElement[]
  readonly bottom: ReactElement[]
}
