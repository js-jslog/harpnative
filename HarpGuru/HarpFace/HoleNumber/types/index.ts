import type { ViewStyle, TextStyle } from 'react-native'

import type { Coord } from '../../HarpRow'

export enum HoleNumberIds {
  One = '1'      , Two = '2'      , Three = '3'   , Four = '4',
  Five = '5'     , Six = '6'      , Seven = '7'   , Eight = '8',
  Nine = '9'     , Ten = '10'     , Eleven = '11' , Twelve = '12',
  Thirteen = '13', Fourteen = '14', Fifteen = '15', Sixteen = '16',
}

export type HoleNumberProps = {
  readonly xCoord: Coord;
}

export type HoleNumberStyles = {
  readonly cell: ViewStyle;
  readonly text: TextStyle;
}
