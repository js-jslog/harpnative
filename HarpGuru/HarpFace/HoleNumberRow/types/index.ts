import type { ViewStyle } from 'react-native'

import type { HarpFaceFragmentProps } from '../../HarpFaceFragment'

export type HoleNumberRowProps = Pick<HarpFaceFragmentProps, 'xRange'>

export type HoleNumberRowStyles = {
  readonly row: ViewStyle
}
