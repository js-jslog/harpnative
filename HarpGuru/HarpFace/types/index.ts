import type { HarpFaceFragmentProps } from '../HarpFaceFragment'

export enum DisplayModes {
  Degree = 'DEGREE',
  Pitch = 'PITCH',
}

export type HarpFaceProps = Omit<HarpFaceFragmentProps, 'xRange'>
