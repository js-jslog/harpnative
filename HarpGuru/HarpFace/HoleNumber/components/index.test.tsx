import React from 'react'
import { render } from '@testing-library/react-native'

import { HoleNumberIds } from '../types'

import { HoleNumber } from './index'

test('HoleNumber renders a dom element with the expected value included', () => {
  const { getByText } = render(<HoleNumber xCoord={0} />)
  expect(getByText(HoleNumberIds.One)).toBeTruthy()
})

test('A snapshot of HoleNumbers', () => {
  const { container } = render(<HoleNumber xCoord={0} />)
  expect(container).toMatchSnapshot()
})
