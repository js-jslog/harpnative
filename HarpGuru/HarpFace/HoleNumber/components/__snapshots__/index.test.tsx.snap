// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A snapshot of HoleNumbers 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 0,
        "justifyContent": "center",
        "width": 31.9374,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#495057",
          "fontSize": 7.5394000000000005,
        }
      }
    >
      1
    </Text>
  </View>
</View>
`;
